pkg/storage/stores/series/series_index_store.go

```diff
@@ -305,7 +305,7 @@ func (c *IndexReaderWriter) chunksToSeries(ctx context.Context, in []logproto.Ch
 		return nil, err
 	}
 
-	results := make([]labels.Labels, len(chunksBySeries)) // Flatten out the per-job results.
+	results := make([]labels.Labels, 0, len(chunksBySeries)) // Flatten out the per-job results.
```

Comment:
this also seems safe, but actually I think it should be `len(perJobResults)` should it not? @bboreham I'm not sure from your original PR ([here](https://github.com/grafana/loki/pull/10310/files)) if you intended to create the slice of `chunkBySeries` length rather than `perJobResults` length

Reply:
No, `len(chunksBySeries)` looks correct. I also feel `len(perJobResults)` is correct at first sight.
Actually, argument of `append` is `innerSlice...`, not `innerSlice`. And sum of their length is probably `len(chunkBySeries)` (`groups := make([]chunkGroup, 0, len(chunksBySeries)/c.chunkBatchSize+1)`.

Reply:
Ah you're right, I misread the append line :+1: good find